tests/unit/builtin_monitors/kubernetes_monitor_test.py

```diff
@@ -1267,3 +1271,195 @@ def mock_k8s_cache_pod(
             result["cont-1"]["k8s_info"]["pod_info"].annotations,
             {"log.config.scalyr.com/attributes.parser": "test-5"},
         )
+
+
+class TestKubernetesKubeletApiAuthTokenCaching(TestConfigurationBase, ScalyrTestCase):
+    @skipIf(platform.system() == "Windows", "Skipping Linux only tests on Windows")
+    def test_re_read_token_every_x_seconds(self):
+        _, token_file_path = tempfile.mkstemp()
+
+        with open(token_file_path, "w") as fp:
+            fp.write("token1")
+
+        self._write_file_with_separator_conversion(
+            """ {
+            api_key: "hi there",
+            k8s_service_account_token: "%s"
+            k8s_token_re_read_interval: 1,
+          }
+        """
+            % (token_file_path)
+        )
+        global_config = self._create_test_configuration_instance()
+        global_config.parse()
+
+        k8s = KubernetesApi.create_instance(global_config=global_config)
+
+        # 1. On initial read we should read first token
+        self.assertEqual(k8s.token, "token1")
+        self.assertEqual(k8s.token, "token1")
+        self.assertEqual(k8s.token, "token1")
+
+        # Update the token
+        with open(token_file_path, "w") as fp:
+            fp.write("token2")
+
+        self.assertEqual(k8s.token, "token1")
+
+        # 2. After the token has been updated, the code should eventually re-read a new token
+        time.sleep(1.2)
```

Comment:
I guess there're no easy, non-invasive ways to make API instances read tokens with a smaller interval. Just in case we can reduce the overall time for testing :)

Reply:
Yeah, I was actually thinking about the same think, but then decided to not do it since it's only couple of seconds.

But since you also mentioned it, I decided I will make the change - I will change the option type from int to float and use a shorted interval (e.g. 0.1s).

Reply:
EDIT: I just tested it locally and we don't save more than 2 seconds - I still need to keep the sleep to be at least 0.5-0.6 seconds to avoid false positives.

One option would be to replace sleep with counter based mock approach, but probably an overkill for now since it only adds 4 seconds to the test run.

Reply:
I don't see any problem with 4 seconds, especially with the fact that we don't add tests with such sleep so often.